### ISSUE TITLE
I can't run machine

### ISSUE THREAD

**Issue by vinniec2** [ivan-hc/VirtualBox-appimage#12]
I know I was too quick, maybe it's a case of telepathy, however I just tried virtualbox appimage and it doesn't work.
The interface starts but after I configure a machine and try to start it I get this error:
```
VM Name: nemex

Required capability 'KVM_CAP_KVM_CAP_XCRS' is missing! (VERR_NEM_MISSING_FEATURE).
Driverless mode (VERR_SUP_DRIVERLESS).
Result Code:
NS_ERROR_FAILURE (0X80004005)
Component:
ConsoleWrap
Interface:
IConsole {6ac83d89-6ee7-4e33-8ae6-b257b2e81be8}
```

I use [Nemesis](https://forum.porteus.org/viewtopic.php?f=137&t=11352), which is a derivative of artix, which is a derivative of arch.
So I opened the [arch explanation page on kvm](https://wiki.archlinux.org/title/KVM) and it looks like I have kvm present.
If you want me to do some testing let me know.

**Comment by vinniec2** [comment]
Ok, seems that it's [my problem](https://github.com/cyberus-technology/virtualbox-kvm/issues/4):

[My processor](https://www.intel.com/content/www/us/en/products/sku/47554/intel-core-i5520um-processor-3m-cache-1-06-ghz/specifications.html) is an "Arrandale" and not an "Sandy Bridge".
I cant use KVM version of Vbox :(

**Comment by ivan-hc** [comment]
its hard to export a virtual machines manager in AppImage format, and the Vitrtualbox-KVM team at https://github.com/cyberus-technology/virtualbox-kvm is a bless in this sense. At least we have not the "vboxdrv" dependence, but still ve have to use "sudo" to enable USB support.

I'm not an expert of CPUs, normally I need to enable virtualization from EFI or BIOS. I'd like you figure out where is the issue.

**Comment by vinniec2** [comment]
Yes, by consulting the issue I had found https://github.com/cyberus-technology/virtualbox-kvm/issues/4 it seems that my processor is missing the “xsave” instruction that is needed, so I guess with all the effort in the world there is no way to help me without changing processors.
Anyway, I have another computer, now I don't remember what processor it has, but if I can try it (since I need virtualbox) surely I will put here the result

**Comment by ivan-hc** [comment]
ok, let me know

**Comment by vinniec2** [comment]
It gave me the same error, I honestly thought that [this processor](https://ark.intel.com/content/www/us/en/ark/products/91832/intel-celeron-processor-n3060-2m-cache-up-to-2-48-ghz.html) was also too old, however, it seems not, [it seems fifth generation](https://en.wikipedia.org/wiki/Next_Unit_of_Computing#Braswell) so I don't know...

I don't understand much about processors unfortunately... maybe I need to try asking in that other issue

**Comment by ivan-hc** [comment]
I have an AMD Ryzen 5 and I have never faced issues. Before I had an AMD Athlon 2, and no issue at all. About Intel, I don't know how thay manage virtualization. Have you tried with the native virtualbox package for your distro?

**Comment by vinniec2** [comment]
With other distros yes and it worked, with this one specifically I have a bit of a problem because I would have to compile the kernel since it is not the same one provided by arch (a bit messy).

**Comment by ivan-hc** [comment]
If I'm not wrong, Artix is a non-systemd distro, have you had same issues on a pure Artix installation?

I have never heard about Nemesis (but from what I understand it's a name = a program)

**Comment by vinniec2** [comment]
I have not used Artix directly however yes it is an Arch without systemd (with all the compatibility problems that come with it).
Nemesis is a version of porteus, I don't think it has anything to do with kali (porteus itself is a derivative of slackware, Nemesis instead is based on artix).
Anyway, yes, the appimages run more or less without problems, among yours I am using the gimp, instead from your colleague I am using the appimage of pavucontrol.

**Comment by ivan-hc** [comment]
virtualbox appimage is quite different, it uses bubblewrap to mount common mountpoints available in other distros:

this come from my fork of Conty https://github.com/ivan-hc/Conty/blob/master/conty-start.sh
```
	launch_wrapper "${bwrap}" \
			"${bind_root[@]}" \
			--dev-bind /dev /dev \
			--ro-bind /sys /sys \
			--bind-try /tmp /tmp \
			--proc /proc \
			--bind-try /home /home \
			--bind-try /mnt /mnt \
			--bind-try /media /media \
			--bind-try /run /run \
			--bind-try /var /var \
			--bind-try /usr/lib/locale /usr/lib/locale \
			--ro-bind-try /usr/share/steam/compatibilitytools.d /usr/share/steam/compatibilitytools.d \
			--ro-bind-try /etc/resolv.conf /etc/resolv.conf \
			--ro-bind-try /etc/hosts /etc/hosts \
			--ro-bind-try /etc/nsswitch.conf /etc/nsswitch.conf \
			--ro-bind-try /etc/passwd /etc/passwd \
			--ro-bind-try /etc/group /etc/group \
			--ro-bind-try /etc/machine-id /etc/machine-id \
			--ro-bind-try /etc/asound.conf /etc/asound.conf \
			--ro-bind-try /etc/localtime /etc/localtime \
			"${non_standard_home[@]}" \
			"${sandbox_params[@]}" \
			"${custom_home[@]}" \
			"${mount_opt[@]}" \
			"${xsockets[@]}" \
			"${unshare_net[@]}" \
			"${set_vars[@]}" \
			--setenv PATH "${CUSTOM_PATH}" \
			"${command_line[@]}"
```
plus another mountpoint I added here https://github.com/ivan-hc/VirtualBox-appimage/blob/main/virtualbox-conty-builder.sh
```
CONTY_CMD="${HERE}/conty.sh --bind-try /usr/share/fonts /usr/share/fonts"
```
sometime is enough to create an empty directory on the system, if one of the above is not in place on the host system

take a look to the mountpoints above

**Comment by ivan-hc** [comment]
what I call "Archimages" are Arch Linux containers into AppImages, and also my GIMP is one of these (also if built differently, and without hardware accelleration support... only Conty-based Archimages have this, and my virtualbox, steam, gnome-boxes and bottles are all built using the above mountpoints, see my fork of Conty here https://github.com/ivan-hc/Conty)

**Comment by vinniec2** [comment]
Conty is a replacement for appimage?

**Comment by ivan-hc** [comment]
Conty is contained into that AppImage

if you extract the AppImage `./*AppImage --appimage-extract`, run the script "conty.sh" like this: `./conty.sh virtualbox`

**Comment by ivan-hc** [comment]
I close this for inactivity
